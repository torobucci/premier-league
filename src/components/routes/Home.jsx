import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import photo from '../assets/skysports.png';

const Home = () => {
  const { league } = useSelector((state) => state.league);
  // console.log(league.leng);
  // if (league.length === 0) {
  //   return (
  //     <div className="spinner-container">
  //       <div className="spinner" />
  //       <div className="loading-text">Loading...</div>
  //     </div>
  //   );
  // }
  return (
    <>
      <div className="heading-content">
        <img
          src={photo}
          alt="league stars"
          className="world-image"
        />
        <div className="league-details">
          <h2>Season: 2022/2023</h2>
          <p>Teams: 20</p>
        </div>
      </div>
      <div className="stat-divide">
        <p>Stats by points</p>
      </div>
      <div className="contents">
        {league.length === 0
            && (
            <div className="spinner-container">
              <div className="spinner" />
              <div className="loading-text">Loading...</div>
            </div>
            )}
        {league.map((team) => (
          <NavLink to={`/team/${team.Name}`} className="details-link" key={team.Name}>
            <div className="cont">
              <div className="img-ico">
                <img src={team.SquadLogo} alt="team logo" className="flags" />
                <i className="fa-solid fa-circle-arrow-right" />
              </div>
              <div className="text-desc">
                <p className="team-name">{team.Name}</p>
                <p className="points">{`${team.Points} pts`}</p>
              </div>
            </div>
          </NavLink>
        ))}
      </div>
    </>

  );
};

export default Home;
