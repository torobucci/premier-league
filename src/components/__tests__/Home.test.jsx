import React from 'react';
import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Home from '../routes/Home';
import '@testing-library/jest-dom';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

describe('Home component', () => {
  beforeEach(() => {
    useSelector.mockClear();
  });

  test('renders without crashing', () => {
    useSelector.mockReturnValue({ league: [] });
    render(<Home />);
  });

  test('displays the correct image', () => {
    useSelector.mockReturnValue({ league: [] });
    const { getByAltText } = render(<Home />);
    const image = getByAltText('league stars');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', 'skysports.png');
  });

  test('displays the correct heading details', () => {
    const mockLeague = [];
    useSelector.mockReturnValue({ league: mockLeague });
    const { getByText } = render(<Home />);
    expect(getByText('Season: 2022/2023')).toBeInTheDocument();
    expect(getByText('Teams: 20')).toBeInTheDocument();
  });

  test('displays the correct stats by points text', () => {
    useSelector.mockReturnValue({ league: [] });
    const { getByText } = render(<Home />);
    expect(getByText('Stats by points')).toBeInTheDocument();
  });

  test('displays correct details for each team', () => {
    const mockLeague = [
      {
        Name: 'Team A',
        SquadLogo: 'team-a-logo.png',
        Points: 50,
      },
      {
        Name: 'Team B',
        SquadLogo: 'team-b-logo.png',
        Points: 40,
      },
    ];
    useSelector.mockReturnValue({ league: mockLeague });
    const { getByText } = render(<BrowserRouter><Home /></BrowserRouter>);

    mockLeague.forEach((team) => {
      expect(getByText(team.Name)).toBeInTheDocument();
      expect(getByText(`${team.Points} pts`)).toBeInTheDocument();
    });
  });
  test('renders correctly', () => {
    const mockLeague = [
      {
        Name: 'Team A',
        SquadLogo: 'team-a-logo.png',
        Points: 50,
      },
      {
        Name: 'Team B',
        SquadLogo: 'team-b-logo.png',
        Points: 40,
      },
    ];
    useSelector.mockReturnValue({ league: mockLeague });
    const { container } = render(<BrowserRouter><Home /></BrowserRouter>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
