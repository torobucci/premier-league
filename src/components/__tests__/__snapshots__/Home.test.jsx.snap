// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<div
  class="heading-content"
>
  <img
    alt="league stars"
    class="world-image"
    src="skysports.png"
  />
  <div
    class="league-details"
  >
    <h2>
      Season: 2022/2023
    </h2>
    <p>
      Teams: 20
    </p>
  </div>
</div>
`;
